### ISSUE TITLE
No type checking error for missing required keyword parameter if no parameters are provided 

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#2771]
### I tried this:

```js
bring "@cdktf/provider-aws" as aws;

new aws.dynamodbTable.DynamodbTable();
```

### This happened:

No compiler error

### I expected this:

A compiler error since the field `name` is expected.

Notice that if I provide one of the optional fields:

```js
bring "@cdktf/provider-aws" as aws;

new aws.dynamodbTable.DynamodbTable(
  hashKey: "id",
);
```

Then the correct error message does surface:

```js
error: Missing required field "name" from "DynamodbTableConfig"
  --> new_dynamo.w:3:1
  |  
3 | / new aws.dynamodbTable.DynamodbTable(
4 | |   hashKey: "id",
5 | | );
  | \-^ Missing required field "name" from "DynamodbTableConfig"
```

### Component

Compiler

### Wing Version

0.18.7

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by staycoolcall911** [comment]
@Chriscbr - in your topmost code sample here - you meant to not pass the `name` property, right?
Also, could you please provide a direction for implementing the fix? This will greatly help someone else implement this, thank you!

**Comment by Chriscbr** [comment]
Fixed the description, thanks!

I'm not sure exactly where the bug lies but the first place I might look is how we type check "new" expressions here:

https://github.com/winglang/wing/blob/8b688a6c82456028d847c803ca0d2ff697058066/libs/wingc/src/type_check.rs#L1297-L1302

**Comment by staycoolcall911** [comment]
This is fixed in #4300
